### PR TITLE
Start developer docs for onboarding a state

### PR DESCRIPTION
**Pull request by echappen**
Also adds iac script to generate service principle for a state storage account.

Addresses #576

### CODE REVIEW

File: docs/onboarding-a-state.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Onboarding a state
+> ⚠️ This documentation is for Piipan developers only
+
+1. Create a service principle for state storage account
```
Comment (heymatthenry) [inline docs/onboarding-a-state.md:4]: Small thing, but I believe it's [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals#service-principal-object:~:text=Service%20principal%20object,-To).

Comment (echappen) [reply docs/onboarding-a-state.md:4]: 🤦‍♀️ this has been a lifelong confusion for me

File: iac/create-state-storage-service-principal.bash [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  # Get resource id of storage account
+  scope=`az resource show -g $RESOURCE_GROUP -n $storage_acct_name --resource-type "Microsoft.Storage/storageAccounts" --query id --output tsv`
+
+  # If the service principal does not exist, the `create-for-rbac` command will
+  # create it. If the service principal does exist, `create-for-rbac` will "patch"
+  # it, creating new credentials.
+  echo "Creating/resetting service principal $name"
+  az ad sp create-for-rbac \
+    --name $name \
+    --role ba92f5b4-2d11-453d-a403-e96b0029c9fe \
```
Comment (ryanhofdotgov) [inline iac/create-state-storage-service-principal.bash:41]: magic number nitpick: doesn't hold up anything, and I know it's in the script preamble, but I'd throw this in a constant with a comment on what it is.

**Review by ryanhofdotgov — APPROVED** [review]
Nice – Thanks for whipping this up!

**Pull request merged**